packages/@ember/-internals/glimmer/lib/utils/bindings.ts

```diff
@@ -118,47 +127,88 @@ export const AttributeBinding = {
 const DISPLAY_NONE = 'display: none;';
 const SAFE_DISPLAY_NONE = htmlSafe(DISPLAY_NONE);
 
-class StyleBindingReference extends CachedReference<string | SafeString> {
-  public tag: Tag;
-  constructor(private inner: Reference<string>, private isVisible: Reference<Opaque>) {
-    super();
+let StyleBindingReference:
+  | undefined
+  | { new (...args: any[]): CachedReference<string | SafeString> };
+
+if (EMBER_COMPONENT_IS_VISIBLE) {
+  StyleBindingReference = class extends CachedReference<string | SafeString> {
+    public tag: Tag;
+    private component: Component;
+    constructor(
+      private inner: Reference<string>,
+      private isVisible: Reference<Opaque>,
+      component: Component
+    ) {
+      super();
+
+      this.tag = combine([inner.tag, isVisible.tag]);
+      this.component = component;
+    }
 
-    this.tag = combine([inner.tag, isVisible.tag]);
-  }
+    compute(): string | SafeString {
+      let value = this.inner.value();
+      let isVisible = this.isVisible.value();
+
+      if (isVisible !== undefined) {
+        deprecate(
+          `\`isVisible\` is deprecated (from "${this.component._debugContainerKey}")`,
+          false,
+          {
+            id: 'ember-component.is-visible',
+            until: '4.0.0',
+            url: 'https://deprecations.emberjs.com/v3.x#toc_ember-component-is-visible',
+          }
+        );
+      }
 
-  compute(): string | SafeString {
-    let value = this.inner.value();
-    let isVisible = this.isVisible.value();
+      if (isVisible !== false) {
+        return value;
+      }
 
-    if (isVisible !== false) {
-      return value;
-    } else if (!value) {
-      return SAFE_DISPLAY_NONE;
-    } else {
-      let style = value + ' ' + DISPLAY_NONE;
-      return isHTMLSafe(value) ? htmlSafe(style) : style;
+      if (!value) {
+        return SAFE_DISPLAY_NONE;
+      } else {
+        let style = value + ' ' + DISPLAY_NONE;
+        return isHTMLSafe(value) ? htmlSafe(style) : style;
+      }
     }
-  }
+  };
 }
 
-export const IsVisibleBinding = {
-  install(_element: Simple.Element, component: Component, operations: ElementOperations) {
-    operations.setAttribute(
-      'style',
-      map(referenceForKey(component, 'isVisible'), this.mapStyleValue),
-      false,
-      null
-    );
-    // // the upstream type for addDynamicAttribute's `value` argument
-    // // appears to be incorrect. It is currently a Reference<string>, I
-    // // think it should be a Reference<string|null>.
-    // operations.addDynamicAttribute(element, 'style', ref as any as Reference<string>, false);
-  },
+export let IsVisibleBinding: any;
```

Comment:
`any` is a bit unfortunate here, can we make it:

```ts
export let IsVisibleBinding: undefined | {
  install(element: Simple.Element, component: Component, operations: ElementOperations): void;
};
```

(I think `mapStyleValue` is only used internally?)

Reply:
Thanks, I've updated it to:

```js
export let IsVisibleBinding: undefined | {
  install(element: Simple.Element, component: Component, operations: ElementOperations): void;
  mapStyleValue(isVisible: boolean, component: Component): SafeString | null;
};
```

With my currently limited knowledge of TS, I couldn't find a way to mark `mapStyleValue` as private.